kedro/runner/runner.py

```diff
@@ -166,25 +180,97 @@ def run_only_missing(
 
         return self.run(to_rerun, catalog, hook_manager)
 
+    @abstractmethod  # pragma: no cover
+    def _get_executor(
```

Comment:
Could we make this less specific and just return `Executor`? Both executors have the same interface and it would be nice if we just program against the interface rather than the concretes. That applies to all other places where for type hints we could just set the type as `Executor` and not the concrete implementations.

https://docs.python.org/3/library/concurrent.futures.html#executor-objects

Reply:
Done!